cmd/dependabot/internal/cmd/update.go

```diff
@@ -156,6 +85,86 @@ var updateCmd = &cobra.Command{
 	},
 }
 
+func extractInput(cmd *cobra.Command) (*model.Input, []byte, error) {
```

Comment:
This function seems to be trying to do too many things, as it's both parsing data (including error checks) but only for some input sources (whereas other input sources have their own dedicated functions) and also contains the logic of which input takes precedence.

Would it work to split "processing each input source" into a dedicated function? That doesn't have to concern itself with data coming from other input sources?

For mere mortals like me, it's a little simpler if functions have a main "area of responsibility" so that if I'm in an area, I'm mentally thinking "this function handles parsing data from X input source" or "this function is the decision tree on what wins out"...